stdlib/public/core/StringObject.swift

```diff
@@ -852,7 +877,12 @@ extension _StringObject {
     }
     return object
 #else
-    _internalInvariant(largeIsCocoa && !isImmortal)
+    _internalInvariant((largeIsCocoa && !isImmortal) || isImmortalSharedCocoa)
+#if arch(arm64)
+    if isImmortalSharedCocoa {
```

Comment:
Definitely add a comment as to why this one form is treated specially rather than all the others. Explain why everywhere else we strip the top nibble but not here.